### PR TITLE
Expanded testing protocol

### PR DESCRIPTION
**Pull request by a11r**
Added the capability to request specific compression types and specific status codes.

### CODE REVIEW

**Comment by a11r** [comment]
@dgquintas PTAL for the definition of CompressionType in messages.proto

**Comment by ejona86** [comment]
@a11r LGTM. I'd be interested where we are going with `CompressionType`, but I expect to learn that later.

**Comment by ejona86** [comment]
I'll let @dgquintas take a look, and then one of us can merge?

File: test/proto/test.proto [hunk 1]
```diff
@@ -71,3 +71,11 @@ service TestService {
   rpc HalfDuplexCall(stream StreamingOutputCallRequest)
       returns (stream StreamingOutputCallResponse);
 }
+
+
+// A simple service NOT implemented at servers so clients can test for
+// that case.
+service UnImplementedService {
```
Comment (dgquintas) [inline test/proto/test.proto]: total nit: shouldn't we take "unimplemented" as a single word and thus remove the capital I?

**Comment by dgquintas** [comment]
LGTM.

**Comment by a11r** [comment]
Verified with nicholasnoble that the "Unstable & flaky tests" breakage can be ignored safely as that check is completely disabled now.

File: test/proto/test.proto (hunk 1 above)
Comment (ejona86) [inline test/proto/test.proto]: +1

**Pull request merged**